### PR TITLE
feat: add mjs support

### PR DESCRIPTION
**Pull request by rishabh3112**
**What kind of change does this PR introduce?**
Support for `.mjs` configs

**Did you add tests for your changes?**

**If relevant, did you update the documentation?**

**Summary**
It is extension of @anshumanv's PR with support for esm configs using `esm` package.

**Does this PR introduce a breaking change?**

**Other information**
The standard `await import()` will not work as https://github.com/nodejs/node/pull/32985 has been reverted.
Using `esm` package in this PR to have esm interop

### CODE REVIEW

File: test/config-format/mjs/webpack.config.mjs [hunk 1]
```diff
@@ -0,0 +1,15 @@
+// import { dirname, resolve } from 'path';
+// import { fileURLToPath } from 'url';
+
```
Comment (evenstensberg) [inline test/config-format/mjs/webpack.config.mjs:3]: remove these

File: test/config-format/mjs/webpack.config.mjs [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+// const __dirname = dirname(fileURLToPath(import.meta.url));
+
+const config = {
```
Comment (evenstensberg) [inline test/config-format/mjs/webpack.config.mjs:6]: export default { ... }

**Review by evenstensberg — CHANGES_REQUESTED** [review]
Tests are failing

**Comment by rishabh3112** [comment]
> Tests are failing

@evenstensberg Yes, because `esm` package is not working properly here. I couldn't find upon investigation on how to make it work.
So, I need help.

**Comment by evilebottnawi** [comment]
I think we don't need package, we can use this idea https://github.com/gulpjs/gulp-cli/blob/master/lib/shared/require-or-import.js

**Comment by rishabh3112** [comment]
> I think we don't need package, we can use this idea https://github.com/gulpjs/gulp-cli/blob/master/lib/shared/require-or-import.js

Same not working again locally.

**Comment by evilebottnawi** [comment]
@rishabh3112 what is node you are use (version)?

**Comment by rishabh3112** [comment]
Tried on 12, 13 and 14. (I use fnm)

**Comment by evilebottnawi** [comment]
Maybe something broken in `fnm`? I use `nvm` and tests for gulp-cli worked with mjs

**Comment by rishabh3112** [comment]
> Maybe something broken in fnm? .

Can't say for sure. Lets see CI, if still not passed then can you try checking out this branch and testing locally once?

**Comment by anshumanv** [comment]
I tried with nvm in the past, no luck 😞

**Comment by anshumanv** [comment]
Yep would be great if @evilebottnawi can check once

**Pull request closed without merging**